### PR TITLE
Refactor and pass more informations to BuyButton

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?
A little refactor of the component and pass more information about the sku items to the `BuyButton`.

#### What problem is this solving?
As per the new requirements to the `addToCart` event, these new properties are required on the `BuyButton` component.

#### How should this be manually tested?
[Workspace](https://lucas--storecomponents.myvtex.com/st-tropez-shorts/p).

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/ProductDetails.js [hunk 1]
```diff
@@ -371,4 +396,7 @@ function mergeSchemaAndDefaultProps(schema, propName) {
   })
 }
 
-export default withRuntimeContext(ProductDetails)
+export default hoistNonReactStatic(
```
Comment (Klynger) [inline react/ProductDetails.js]: [withRuntimeContext](https://github.com/vtex-apps/render-runtime/blob/9c1276a99ef7bb43de684f16fd374c08c5d43ff4/react/components/RenderContext.tsx#L14) already does that, no?

Comment (lucasecdb) [reply react/ProductDetails.js]: yeah, but `injectIntl` does not.

Comment (Klynger) [reply react/ProductDetails.js]: It does too: https://github.com/yahoo/react-intl/blob/master/src/inject.js

Comment (lucasecdb) [reply react/ProductDetails.js]: nice! didn't know it was released already.

**Review by Klynger — APPROVED** [review]

**Review by brunojdo — APPROVED** [review]

**Pull request merged**